backend/app/api/routes/podcasts.py

```diff
@@ -0,0 +1,151 @@
+    pods = session.exec(select(Podcast).where(Podcast.course_id == course_id)).all()
+    return PodcastsPublic(data=[PodcastPublic.model_validate(p) for p in pods])
+
+
+class GeneratePodcastRequest(BaseModel):
+    title: Optional[str] = None
+    mode: Optional[str] = None  # 'dialogue' | 'presentation'
+    topics: Optional[str] = None
+    teacher_voice: Optional[str] = None
+    student_voice: Optional[str] = None
+    narrator_voice: Optional[str] = None
+    document_ids: Optional[List[uuid.UUID]] = None
+
+
+@router.post("/{course_id}/generate", response_model=PodcastPublic)
+async def generate_podcast(
+    course_id: uuid.UUID,
+    session: SessionDep,
+    current_user: CurrentUser,
+    body: GeneratePodcastRequest | None = None,
+) -> Any:
+    if not body or not body.title or not body.title.strip():
+        raise HTTPException(status_code=422, detail="Title is required")
+    title = body.title.strip()
+    mode = (body.mode or "dialogue") if body else "dialogue"
+    topics = body.topics if body else None
+    teacher_voice = body.teacher_voice if body and body.teacher_voice else settings.PODCAST_TEACHER_VOICE
+    student_voice = body.student_voice if body and body.student_voice else settings.PODCAST_STUDENT_VOICE
+    narrator_voice = body.narrator_voice if body and body.narrator_voice else settings.PODCAST_TEACHER_VOICE
+    doc_ids = body.document_ids if body and body.document_ids else None
+    podcast = await generate_podcast_for_course(
+        session,
+        course_id,
+        title,
+        teacher_voice,
+        student_voice,
+        narrator_voice,
+        mode,
+        topics,
+        doc_ids,
+    )
+    return PodcastPublic.model_validate(podcast)
+
+
+@router.get("/by-id/{podcast_id}", response_model=PodcastPublic)
+def get_podcast(podcast_id: uuid.UUID, session: SessionDep, current_user: CurrentUser) -> Any:
+    pod = session.get(Podcast, podcast_id)
+    if not pod:
+        raise HTTPException(status_code=404, detail="Podcast not found")
+    return PodcastPublic.model_validate(pod)
+
+
+@router.get("/by-id/{podcast_id}/audio")
+def stream_audio(podcast_id: uuid.UUID, session: SessionDep, current_user: CurrentUser):
+    pod = session.get(Podcast, podcast_id)
+    if not pod:
+        raise HTTPException(status_code=404, detail="Podcast not found")
+    if pod.storage_backend == "local":
+        file_path = pod.audio_path
+        if not os.path.exists(file_path):
+            raise HTTPException(status_code=404, detail="Audio file missing")
+        def iterfile():
+            with open(file_path, "rb") as f:
+                while chunk := f.read(8192):
+                    yield chunk
+        return StreamingResponse(iterfile(), media_type="audio/mpeg")
+    else:
+        # For S3, return a presigned URL to let client fetch directly
+        try:
+            import boto3
+            s3 = boto3.client(
+                "s3",
+                aws_access_key_id=settings.AWS_ACCESS_KEY_ID,
```

Comment:
Share the S3 credentials with @deluakin so that he can add them to the Render Backend API service.